pipenv/cli/command.py

```diff
@@ -226,7 +226,9 @@ def install(
     **kwargs
 ):
     """Installs provided packages and adds them to Pipfile, or (if no packages are given), installs all packages from Pipfile."""
-    from ..core import do_install
+    from ..core import do_install, load_dot_env
+
+    load_dot_env()
```

Comment:
This change is out of your proposed behavior. You should change it and related module reloading part in a separate PR.
There were already many discussions about whether to respect dot env files in **setup** time because it is now used for **run** time only. #3398 #3080 #3271

Reply:
I removed the loading of the env. Prior discussions aside, this feels inconsistent: if we already set up python via `ensure_python` at other times, we should be loading `.env` those times as well.

As it is, my workaround would be something silly like `pipenv shell -- 'pipenv install; exit;'` to achieve the presence of `.env` files, which I can do, but feels lame.

However, the module reload is important, because without it, the check for `PIPENV_YES` fails during install unless that is set in the environment.

In general it feels like the Pipenv code is playing pretty fast and loose with whether or not it uses `.env` modified env vars: it freezes those in `environments.py` at compile time, but sometimes doesn't `import environments` until runtime, deferring environment reading by ... I can't tell if it's an accident or a very oblique way of re-reading the environment. Either way, I think a reload is better--not great, but better; a real solve here would be a better interface to all the environment vars Pipenv cares about, which I mentioned in another comment.

Followup questions:
- Is removing the `.env` load at install time sufficient?
- If environment module reloading must be removed, should I instead fix the `PIPENV_YES` check?
- If not, what remaining course of action achieves the goal (Pipenv can be set up and installed with nothing particular set in the shell)?